docs/enhancements/boot-status-reporter.md

```diff
@@ -0,0 +1,129 @@
+---
+title: boot-status-reporter
+authors:
+  - "@nmagnezi"
+creation-date: 2022-09-22
+last-updated: 2022-09-22
+---
+
+# Assisted Boot Reporter
+
+## Summary
+
+Create a new reporting service that runs as early as possible when hosts are booted from disk to collect additional 
+information and send it to `assisted-service`.
+
+## Motivation
+
+There are multiple occurrences of clusters that fail to install at the stage where
+hosts manage to pull ignition but fail to get to a point where the `assisted-controller`
+pod runs and calls home to `assisted-service` with logs.
+
+With that missing information, there's not a whole lot we can do to understand what
+happened with those installations.
+
+
+### Goals
+
+1. Gather information for a host boot stage to improve our ability to debug failures and better inform 
+   users about the host progress stages (first boot, second boot, etc).
+
+
+2. The new service should be thoroughly cleaned up from the hosts and should never fail installations by itself.
+   If the boot reporter service fails to run or report logs, the cluster installation should continue, 
+   and the new host progress stages should be skipped in such a scenario.
+
+## Proposal
+
+
+### Changes To Assisted Service
+For cases as mentioned in `Motivation`, we need a new service (start via `systemd`) soon as possible after hosts
+are booting from the disk and contact `assisted-service` to:
+
+1. Change the installation stage to a new `Booted with local ignition` Between `Configuring`
```

Comment:
I don't like the new stage name too much because it kinda sounds like what Configuring already is. How about `Reporting`?

Reply:
Could we piggy-back here and maybe rethink renaming `Configuring`? I'm not sure we have a common agreement that everyone understands that Configuring means "ignition pulled from MCS but not necessarily applied correctly". Maybe something like following states: `Configuring (ignition pulled)` followed by `Configuring (ignition applied)`?